File: app/src/main/java/com/callmangement/EHR/imagepicker/ui/imagepicker/ImagePickerView.java
package com.callmangement.EHR.imagepicker.ui.imagepicker;



import com.callmangement.EHR.imagepicker.model.Folder;
import com.callmangement.EHR.imagepicker.model.Image;
import com.callmangement.EHR.imagepicker.ui.common.MvpView;

import java.util.List;

/**
 * Created by hoanglam on 8/17/17.
 */

public interface ImagePickerView extends MvpView {

    void showLoading(boolean isLoading);

    void showFetchCompleted(List<Image> images, List<Folder> folders);

    void showError(Throwable throwable);

    void showEmpty();

    void showCapturedImage();

    void finishPickImages(List<Image> images);

}

File: app/src/main/java/com/callmangement/ehr/imagepicker/ui/imagepicker/ImagePickerView.kt
package com.callmangement.ehr.imagepicker.ui.imagepicker

import com.callmangement.ehr.imagepicker.model.Folder
import com.callmangement.ehr.imagepicker.model.Image
import com.callmangement.ehr.imagepicker.ui.common.MvpView


/**
 * Created by hoanglam on 8/17/17.
 */
interface ImagePickerView : MvpView {
    fun showLoading(isLoading: Boolean)

    fun showFetchCompleted(images: List<Image>?, folders: List<Folder>?)

    fun showError(throwable: Throwable?)

    fun showEmpty()

    fun showCapturedImage()

    fun finishPickImages(images: List<Image?>?)
}